main/src/mill/main/AndroidHiltModule.scala

```diff
@@ -0,0 +1,23 @@
+package mill.main
+
+import coursier.LocalRepositories
+import coursier.core.Repository
+import coursier.maven.MavenRepository
+import mill.api.PathRef
+import mill.define.{Discover, ExternalModule, Target}
+import mill.util.MillModuleUtil.millProjectModule
+
+object AndroidHiltModule extends ExternalModule {
+  def repositories: Seq[Repository] = Seq(
+    LocalRepositories.ivy2Local,
+    MavenRepository("https://repo1.maven.org/maven2"),
+    MavenRepository("https://maven.google.com/")
+  )
```

Comment:
This looks suspicious and ignored any user configuration. I assume, you need it to resolve dependencies later.

Instead you could hold a reference to a `JvmWorkerModule` by extending `WithJvmWorker` and use `jvmWorker().defaultResolver()` to resolve dependencies.

Reply:
I've removed the module from main, it didn't make sense to me to be there, and moved it to the hilt module file which I think now resolves the dependencies properly (my tests were also failing because of this)